hack/test/assets/test_release_candidates.yaml

```diff
@@ -0,0 +1,27 @@
+snapshots:
+  - metadata:
+      generated_at: '2025-03-10T10:32:04.642635'
+      status: pending
+      id: "rc-20250310-001"
+    components:
+      - repository: https://github.com/kubernetes-sigs/cluster-api
+        ref: v1.9.5
+        image_url:
+      - repository: https://github.com/metal3-io/cluster-api-provider-metal3
+        ref: v1.9.3
+        image_url:
+      - repository: https://github.com/openshift/assisted-service
+        ref: 76d29d2a7f0899dcede9700fc88fcbad37b6ccca
+        image_url: quay.io/edge-infrastructure/assisted-service:latest-76d29d2a7f0899dcede9700fc88fcbad37b6ccca
+      - repository: https://github.com/openshift/assisted-image-service
+        ref: 2249c85d05600191b24e93dd92e733d49a1180ec
+        image_url: quay.io/edge-infrastructure/assisted-image-service:latest-2249c85d05600191b24e93dd92e733d49a1180ec
+      - repository: https://github.com/openshift/assisted-installer-agent
+        ref: cfe93a9779dea6ad2a628280b40071d23f3cb429
+        image_url: quay.io/edge-infrastructure/assisted-installer-agent:latest-cfe93a9779dea6ad2a628280b40071d23f3cb429
+      - repository: https://github.com/openshift/assisted-installer
+        ref: c389a38405383961d26191799161c86127451635
+        image_url: quay.io/edge-infrastructure/assisted-installer-controller:latest-c389a38405383961d26191799161c86127451635
```

Comment:
Thinking about image, I think we should definitely have the image repository (i.e. `quay.io/edge-infrastructure/assisted-installer-controller`), so that we can generalize how to search the latest image.
We might want to save the full image with digest/tag in "tested_image" or something like that